src/ConfigurationClientWrapper.ts

```diff
@@ -12,7 +12,7 @@ export class ConfigurationClientWrapper {
     endpoint: string;
     client: AppConfigurationClient;
     backoffEndTime: number = 0; // Timestamp
-    #failedAttempts: number = 0;
+    failedAttempts: number = 0;
```

Comment:
If you have to expose failedAttempt to be public for testing, you should make this private and create another public property which only has getter. And the getter will return the private #failedAttempt. This can make sure failedAttempt won't be modified outside of the class.